### PR TITLE
feat: add Bilibili OAuth

### PR DESCRIPTION
**Pull request by jiak94**

### CODE REVIEW

File: idp/provider.go [hunk 1]
```diff
@@ -86,6 +86,8 @@ func GetIdProvider(typ string, subType string, clientId string, clientSecret str
 		return NewCasdoorIdProvider(clientId, clientSecret, redirectUrl, hostUrl)
 	} else if isGothSupport(typ) {
 		return NewGothIdProvider(typ, clientId, clientSecret, redirectUrl)
+	} else if typ == "Bilibili" {
+		return NewBilibiliIdProvider(clientId, clientSecret, redirectUrl)
```
Comment (hsluoyz) [inline idp/provider.go:90]: @jiak94 thanks for this PR! BTW, the frontend JS code also needs changes to allow users to choose this new provider, see previous PRs about how to do it: https://github.com/casdoor/casdoor/pulls?q=is%3Apr+is%3Aclosed+provider

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: idp/provider.go (hunk 1 above)
Comment (jiak94) [reply idp/provider.go:90]: will update the PR later this week.

Comment (hsluoyz) [reply idp/provider.go:90]: @jiak94 hi, any update?

**Comment by hsluoyz** [comment]
@jiak94 plz resolve conflicts

image

**Review by hsluoyz — APPROVED** [review]

File: conf/app.conf [hunk 2]
```diff
@@ -4,7 +4,7 @@ runmode = dev
 SessionOn = true
 copyrequestbody = true
 driverName = mysql
-dataSourceName = root:123456@tcp(localhost:3306)/
+dataSourceName = root:091121@tcp(localhost:3306)/
```
Comment (hsluoyz) [inline conf/app.conf:7]: @jiak94  Don't change this and `yarn.lock`

**Comment by hsluoyz** [comment]
@jiak94 can you also help add it to docs? https://casdoor.org/docs/provider/oauth/baidu

**Pull request merged**